pkg/otlp/collector.go

```diff
@@ -154,13 +163,17 @@ type CollectorStatus struct {
 }
 
 // NewPipeline defines a new OTLP pipeline.
-func NewPipeline(cfg PipelineConfig, s serializer.MetricSerializer) (*Pipeline, error) {
+func NewPipeline(cfg PipelineConfig, s serializer.MetricSerializer, logsAgentChannel chan *message.Message) (*Pipeline, error) {
 	buildInfo, err := getBuildInfo()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get build info: %w", err)
 	}
 
-	factories, err := getComponents(s)
+	if cfg.LogsEnabled && logsAgentChannel == nil {
```

Comment:
Is there a test case?

Reply:
Added in `collector_test.go`

Reply:
Can you add a test yaml config to `pkg/otlp/testdata` and a test case in https://github.com/DataDog/datadog-agent/blob/main/pkg/otlp/config_test.go#L19?